### PR TITLE
[WIP] dasharo-compatibility-test-specification 

### PR DESCRIPTION
**Pull request by sulewskiprzemyslaw**
Signed-off-by: Przemyslaw Sulewski przemyslaw.sulewski@3mdeb.com

### CODE REVIEW

**Comment by artur-rs** [comment]
@pietrushnic this is initial approach to describe/gather unified tests specification in the Dasharo ecosystem.

Our questions:
1. Files structure - maybe you have another vision where wew should add those docs
2. Test matrix: which parameter is best suited to categorize the tests
  * processors
  * architectures
  * vendor platform name / code name, like apu1
  * dasharo product lines, like Dasharo Workstation

**Comment by pietrushnic** [comment]
Feel free to reuse text that I will write below. I think we should start gather vision of Dasharo in once place and eliminate already outdated concepts.

First thing, SKU, flavors and modules are rather outdated concepts from the beginning phase of the project. Despite that I like division of tests by module, suite and case. Recent Dasharo releases show what customers and community want and those are features, if features are validated then things look good. Test results are not only for us Open Source Firmware Vendor, but also for developers who wants to use code, and end users who want to decide about which version of firmware they want to use.

We should distinguish features by:
* availability - available in upstream open source ecosystem or only in Dasharo, there may be also partial situation, where upstream for various reasons is not fully functional out of the box and additional action by user is needed
* support level - not all features can be validated everywhere, some are visible only on firmware level, some are visible on OS level and some are visible on both levels

other_support

To answer questions:
1. Documentation always can be changed, so let's focus on content rather then structure. Good structure need time to mature
2. This is relate to above description - what given community group needs? Tests should be presented as easy for users as possible. What users care? Initial deployment (aka Installation manual) - what deployment method works for my platform, firmware update - how to update my firmware, recovery (de-bricking) - how to recover my platform, on what systems I can perform those operation and how, if there was stock BIOS can I safely recover to it - those are probably tests to cover in Dasharo flashing module, which should be renamed to something more aligned.

I know we have tests that prove intermediate steps, and it is ok to present results of those, but most important for end users is overall if given feature works, so we should have some simplified view of things.

Please let me know if it answer your questions.

File: docs/dasharo-compatibility-specification/dasharo-compatibility-specification.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+The test matrix decribes, which test cases and test suites are compatible with
+a given device or group of devices. If the given test is compatible with the
+platform, the customer might be sure, that before releasing the new firmware
+version this test was executed and its result might be read from the
+spreadsheet with test results. Also, log from the test might be obtained
+from `release` section.
+
+The test matrix is available under the following 
+[link](https://docs.google.com/spreadsheets/d/1MPbz8OIDuuErZOHdnkxZnUo4snLptQU_mBKat3H2Mow/edit?usp=sharing).
```
Comment (pietrushnic) [inline docs/dasharo-compatibility-specification/dasharo-compatibility-specification.md:52]: Many privacy-conscious users will dislike presenting things in google docs. Also coreboot is always written with small letters :angry:

File: docs/dasharo-compatibility-specification/dasharo-compatibility-specification.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# Dasharo compatibility test specification
+
+Each time the firmware release procedure is preceded by the validation procedure.
+Its main tasks are:
```
Comment (pietrushnic) [inline docs/dasharo-compatibility-specification/dasharo-compatibility-specification.md:4]: Please explain what is the goal of this document in clear and concise way.

File: docs/dasharo-compatibility-specification/dasharo-compatibility-specification.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# Dasharo compatibility test specification
+
+Each time the firmware release procedure is preceded by the validation procedure.
+Its main tasks are: 
+* make sure, that the new firmware meets the requirements related to current
+    stage acceptance criteria,
```
Comment (pietrushnic) [inline docs/dasharo-compatibility-specification/dasharo-compatibility-specification.md:6]:
```suggestion
  stage acceptance criteria,
```

line wrapping here and in below is incorrect please fix that

**Review by pietrushnic — CHANGES_REQUESTED** [review]
I didn't finished the review since there is some background information that have to be explained offline.

**Comment by pietrushnic** [comment]
@artur-rs @sulewskiprzemyslaw when we can get back to discussion here?

**Comment by pietrushnic** [comment]
@artur-rs @sulewskiprzemyslaw ping2

**Comment by artur-rs** [comment]
@sulewskiprzemyslaw please address the review comments. Piotr answered our initial concerns, in my opinion you can continue with the updates

**Comment by sulewskiprzemyslaw** [comment]
@pietrushnic @artur-rs to this topic I will return on Monday, the 7th of march. In addition to making the appropriate corrections I will also do the following things:
- resolve all conflicts,
- add test documentation from the laptops testing (cc @macpijan).

**Pull request closed without merging**